src/core/table/table.jsx

```diff
@@ -0,0 +1,268 @@
+    };
+
+    function descendingComparator(a, b, orderBy) {
+        if (b[orderBy] < a[orderBy]) {
+            return -1;
+        }
+        if (b[orderBy] > a[orderBy]) {
+            return 1;
+        }
+        return 0;
+    }
+
+    function getComparator(order, orderBy) {
+        return order === "desc"
+            ? (a, b) => descendingComparator(a.original, b.original, orderBy)
+            : (a, b) => -descendingComparator(a.original, b.original, orderBy);
+    }
+
+    function stableSort(array, comparator) {
+        const stabilizedThis = array.map((el, index) => [el, index]);
+        stabilizedThis.sort((a, b) => {
+            const order = comparator(a[0], b[0]);
+            if (order !== 0) {
+                return order;
+            }
+            return a[1] - b[1];
+        });
+        return stabilizedThis.map((el) => el[0]);
+    }
+
+    const renderSelectCell = (checked, labelId) => {
+        const multiSelectCell = (
+            <TableCell padding="checkbox">
+                <Checkbox
+                    color="primary"
+                    checked={checked}
+                    inputProps={{
+                        'aria-labelledby': labelId,
+                    }}
+                />
+            </TableCell>
+        );
+
+        const singleSelectCell = (
+            <TableCell padding="checkbox">
+                <Radio
+                    checked={checked}
+                    name="radio-button"
+                    inputProps={{ 'aria-labelledby': labelId }}
+                />
+            </TableCell>
+        );
+        return multiSelect ? multiSelectCell : singleSelectCell;
+    }
+
+    const renderTableBody = () => {
+        return (
+            <TableBody>
+                {stableSort(processedData, getComparator(order, orderBy))
+                    .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+                    .map((row, index) => {
+                        const isItemSelected = isSelected(row._id);
+                        const labelId = `enhanced-table-checkbox-${index}`;
+
+                        return (
+                            <TableRow
+                                hover
+                                onClick={(event) => handleClick(event, row)}
+                                role="checkbox"
+                                aria-checked={isItemSelected}
+                                tabIndex={-1}
+                                key={row.name}
+                                selected={isItemSelected}
+                                sx={{ height: '100px' }}
+                            >
+                                {onRowSelect !== _.noop && renderSelectCell(isItemSelected, labelId)}
+                                {processedColumns.map(column => {
+                                    const Template = column.template;
+
+
+                                    return Template ? (
+                                        <TableCell>
```

Comment:
TableCell could wrap around the actual alternatives

```suggestion
                                        <TableCell>
                                            {Template ? <Template row={row} /> : row.original[column.accessor]}
                                        </TableCell>
```